### PR TITLE
Remove the intrinsics hook, directly call the codegen functions

### PR DESCRIPTION
**Pull request by danielsn**
### Description of changes: 

Currently, intrinsics are resolved using a "hook", which is difficult to understand and debug.
Instead, just use normal function calls to codegen them.
### Resolved issues:

Resolves #ISSUE-NUMBER

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 1]
```diff
@@ -74,7 +120,7 @@ impl<'tcx> GotocCtx<'tcx> {
     /// c.f. rustc_codegen_llvm::intrinsic impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx>
     /// fn codegen_intrinsic_call
     /// c.f. https://doc.rust-lang.org/std/intrinsics/index.html
-    pub fn codegen_intrinsic(
+    pub fn codegen_regular_intrinsic(
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:123]: Can you please document what you mean by regular intrinsic?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 2]
```diff
@@ -39,6 +39,52 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_expr_to_place(p, e)
     }
 
+    pub fn codegen_intrinsic(
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:42]: Can you please document this function too?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 3]
```diff
@@ -39,6 +39,52 @@ impl<'tcx> GotocCtx<'tcx> {
...
+                ],
+                loc,
+            )
+        } else {
+            self.codegen_never_return_intrinsic(instance, Some(span))
+        }
+    }
+
+    /// Returns `Some(instance)` if the function is an intrinsic; `None` otherwise
+    pub fn get_intrinsic_instance(&self, func: &Operand<'tcx>) -> Option<Instance<'tcx>> {
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:66]: Should this be private?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 4]
```diff
@@ -39,6 +39,52 @@ impl<'tcx> GotocCtx<'tcx> {
...
+                    Some(instance)
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        }
+    }
+
+    pub fn is_intrinsic(&self, func: &Operand<'tcx>) -> bool {
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:84]: Please doc

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs (hunk 1 above)
Comment (celinval) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:123]: Can you also please remove `pub`?

**Review by celinval — APPROVED** [review]
Sweet! Can you please address the comments before pushing? Thanks!

File: kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs [hunk 5]
```diff
@@ -390,6 +390,11 @@ impl<'tcx> GotocCtx<'tcx> {
                 }
             })
             .collect();
+
+        if self.is_intrinsic(func) {
```
Comment (tedinski) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs:394]: is `is_instrinsic` just duplicating a bunch of code that's already right below these lines?

Couldn't we just inline the `if matches!...` part of the above function below after the `let instance`?

**Pull request merged**